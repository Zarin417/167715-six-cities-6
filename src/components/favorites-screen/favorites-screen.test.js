import React from "react";
import {render} from "@testing-library/react";
import {createMemoryHistory} from "history";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import FavoritesScreen from "./favorites-screen";
import {Router} from "react-router";

const mockStore = configureStore({});

it(`FavoritesScreen should render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    PLACE_INFO: {placeInfo: {}},
    USER: {isLoggedIn: true},
    FAVORITE: {favorites: [
      {
        title: `The house among olive`,
        price: 169,
        previewImage: ``,
        type: `hotel`,
        isFavorite: true,
        isPremium: true,
        rating: 5,
        id: 15,
        bedrooms: 2,
        description: ``,
        maxAdults: 2,
        location: {},
        images: [``],
        host: {id: 25, name: `Maria`, isPro: true, avatarUrl: `img/avatar-angelina.jpg`},
        goods: [`Breakfast`],
        city: {name: `Amsterdam`, location: {}},
      },
      {
        title: `The good place`,
        price: 269,
        previewImage: ``,
        type: `hostel`,
        isFavorite: true,
        isPremium: true,
        rating: 5,
        id: 25,
        bedrooms: 2,
        description: ``,
        maxAdults: 2,
        location: {},
        images: [``],
        host: {id: 25, name: `Maria`, isPro: true, avatarUrl: `img/avatar-angelina.jpg`},
        goods: [`Breakfast`],
        city: {name: `Cologne`, location: {}},
      },
    ]}
  });
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <FavoritesScreen />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
