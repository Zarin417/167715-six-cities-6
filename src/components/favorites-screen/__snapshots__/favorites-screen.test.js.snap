// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesScreen should render correctly 1`] = `
<div>
  <section
    class="favorites"
  >
    <h1
      class="favorites__title"
    >
      Saved listing
    </h1>
    <ul
      class="favorites__list"
    >
      <li
        class="favorites__locations-items"
      >
        <div
          class="favorites__locations locations locations--current"
        >
          <div
            class="locations__item"
          >
            <a
              class="locations__item-link"
              href="#"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </div>
        </div>
        <div
          class="favorites__places"
        >
          <article
            class="favorites__card place-card"
          >
            <div
              class="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              class="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/15"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="110"
                  src=""
                  width="150"
                />
              </a>
            </div>
            <div
              class="favorites__card-info place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €
                    169
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button
      place-card__bookmark-button--active button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlink:href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width: 100%;"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/15"
                >
                  The house among olive
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                Hotel
              </p>
            </div>
          </article>
        </div>
      </li>
      <li
        class="favorites__locations-items"
      >
        <div
          class="favorites__locations locations locations--current"
        >
          <div
            class="locations__item"
          >
            <a
              class="locations__item-link"
              href="#"
            >
              <span>
                Cologne
              </span>
            </a>
          </div>
        </div>
        <div
          class="favorites__places"
        >
          <article
            class="favorites__card place-card"
          >
            <div
              class="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              class="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/25"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="110"
                  src=""
                  width="150"
                />
              </a>
            </div>
            <div
              class="favorites__card-info place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €
                    269
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button
      place-card__bookmark-button--active button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlink:href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width: 100%;"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/25"
                >
                  The good place
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                Hostel
              </p>
            </div>
          </article>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
